streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -123,15 +126,20 @@
      *         State PARTITIONS_REVOKED may want transit to itself indefinitely, in the corner case when
      *         the coordinator repeatedly fails in-between revoking partitions and assigning new partitions.
      *         Also during streams instance start up PARTITIONS_REVOKED may want to transit to itself as well.
-     *         In this case we will forbid the transition but will not treat as an error.
+     *         In this case we will allow the transition but it will be a no-op as the set of revoked partitions
+     *         should be empty.
      *     </li>
      * </ul>
      */
     public enum State implements ThreadStateTransitionValidator {
-        // TODO: the current transitions from other states directly to PARTITIONS_REVOKED is due to
-        //       the fact that onPartitionsRevoked may not be triggered. we need to refactor the
-        //       state diagram more thoroughly after we refactor StreamsPartitionAssignor to support COOPERATIVE
-        CREATED(1, 5), STARTING(2, 3, 5), PARTITIONS_REVOKED(3, 5), PARTITIONS_ASSIGNED(2, 3, 4, 5), RUNNING(2, 3, 5), PENDING_SHUTDOWN(6), DEAD;
+
+        CREATED(1, 5),                   // 0
```

Comment:
Nothing changed here, just trying to make it more readable. Did modify the fsm diagram above, but only to make it consistent with the transitions listed here -- basically, we can now transition directly to PARTITIONS_ASSIGNED